types/networks/src/lib.rs

```diff
@@ -151,6 +173,34 @@ pub struct ChainConfig {
     pub policy: Policy,
 }
 
+impl ChainConfig {
+    pub fn validate(&self) -> Result<(), String> {
+        let number_of_network_version_variants = std::mem::variant_count::<NetworkVersion>();
+        let number_of_height_variants = std::mem::variant_count::<Height>();
+        if number_of_network_version_variants + 1 != number_of_height_variants {
+            return Err(
```

Comment:
Can we print out which network versions are missing?

Reply:
What do you mean by missing, as we aren't storing network versions?

Reply:
@tyshko5 I think you can just a have a simple loop over all `HEIGHT_VARIANTS` and check if _this_ height is present in the `height_infos` vector of the given config. If not then bail out with an error saying that the config is missing _this_ height.

Reply:
@elmattic How about:
- we default to epoch 0 if an entry is missing from the config, 
- we sort the `height_infos` in `network_version()` such that the order doesn't matter,
- we get rid of `version_schedule` since it should be constant for all networks.

Then we wouldn't have to validate the chain configuration.

Reply:
Yes, I had this in mind too for the third bullet. We shoud remove the `version_schedule` from the config.

Reply:
I'm ok with the sort. But if we are defaulting to 0 for let's say the Skyr upgrade (ie Skyr's epoch is missing in the custom config file), what will happen in case we start importing a nv15 snapshot, then trying to upgrade to nv16?
The goal of config validation to me is to avoid any bad surprises when operating a node.

Reply:
What was the purpose of `Claus` height in the filecoin protocol? We don't really use it anymore in our codebase, should we keep it nonetheless?

Reply:
> What was the purpose of `Claus` height in the filecoin protocol? We don't really use it anymore in our codebase, should we keep it nonetheless?

Nah, let's delete it.

Reply:
@tyshko5 I've updated the original issue with a new description of what needs to be done.